src/Tests/DependencyInjection/CsaGuzzleExtensionTest.php

```diff
@@ -64,6 +64,32 @@ class: AppBundle\Client
         $this->assertEquals('AppBundle\Client', $client->getClass());
     }
 
+    /**
+     * @dataProvider clientConfigInstance
+     * @covers CsaGuzzleExtension::buildGuzzleConfig
+     */
+    public function testClientConfigInstanceOverride($instanceKey)
+    {
+        $serviceId = 'my.message.factory.id';
```

Comment:
Since we're not actually testing the serviceId in a meaningful way, didn't feel the need :?

If it's a deal breaker, I can update the PR :)